ccd-sdk-types/src/main/java/uk/gov/hmcts/ccd/sdk/types/Event.java

```diff
@@ -140,9 +157,42 @@ public boolean isForAllStates() {
       return this;
     }
 
-    EventBuilder<T, R, S> forState(S state) {
-      this.preState = state.toString();
-      this.postState = state.toString();
+    public EventBuilder<T, R, S> forState(S state) {
+      this.preState = List.of(state.getState());
+      this.postState = state.getState();
+      return this;
+
+    }
+    public EventBuilder<T, R, S> initialState(S initialState) {
+      this.preState = List.of();
+      this.postState = initialState.getState();
+      return this;
+    }
+    public EventBuilder<T, R, S> forStateTransition(S from, S to) {
+      this.preState = List.of(from.getState());
+      this.postState = to.getState();
+      this.transition = true;
+      return this;
+    }
+
+    public EventBuilder<T, R, S> forStateTransition(List<S> from, S to) {
```

Comment:
How does the builder code that uses this look? Does requiring a List make it verbose?

We could reverse the order and make them it variadic so you can pass in S literals.

Or maybe use a nested builder so we could do:

.forStateTransition().from(State.X, State.Y).to(State.Z)

Maybe something for another PR however.